src/github.com/mozilla-services/pushgo/simplepush/config.go

```diff
@@ -396,21 +396,21 @@ func LoadApplication(configFile ConfigFile, env envconf.Environment,
 			return serv, nil
 		},
 		PluginSocket: func(app *Application) (HasConfigStruct, error) {
-			h := NewSocketHandler()
+			h := new(SocketHandler)
```

Comment:
Why the switch to new()? The main reason so much code sticks with a function to return it is that way if the initialization requirement changes, you don't have to go find every new(Whatever), you can just change NewWhatever and all the existing code *just works*. Is there some hidden reason for this change?